include/openenclave/internal/posix.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _OE_INTERNAL_POSIX_H
+#define _OE_INTERNAL_POSIX_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/internal/types.h>
+
+/*
+**==============================================================================
+**
+** posix.h
+**
+**     This file defines shared definitions used across the POSIX EDL interface.
+**
+**==============================================================================
+*/
+
+OE_EXTERNC_BEGIN
+
+/*
+**==============================================================================
+**
+** oe_host_fd_t:
+**
+**==============================================================================
+*/
+
+typedef int64_t oe_host_fd_t;
+
+/*
+**==============================================================================
+**
+** oe_register_posix_ocall_function_table():
+** oe_register_posix_ecall_function_table():
+**
+**==============================================================================
+*/
+
+#define OE_POSIX_OCALL_FUNCTION_TABLE_ID 0
+#define OE_POSIX_ECALL_FUNCTION_TABLE_ID 0
+
+/* Register the OCALL table needed by the POSIX interface (host side). */
+void oe_register_posix_ocall_function_table(void);
+
+/* Register the ECALL table needed by the POSIX interface (enclave side). */
+void oe_register_posix_ecall_function_table(void);
+
+/*
+**==============================================================================
+**
+** oe_device_notifications_t:
+**
+**==============================================================================
+*/
+
+OE_PACK_BEGIN
```

Comment:
Can you describe why the struct ought to be packed?

Reply:
Thanks. We cleaned this up and added a comment to explain this is an overlay of a system struct.